students/andykwon/lesson03/mailroom.py

```diff
@@ -0,0 +1,103 @@
+"""
+simple mailroom program for a non profit
+"""
+
+import sys
+
+donors = [("Donor A", [1, 2, 4, 5]),
+          ("Donor B", [14, 5]),
+          ("Donor C", [1, 5, 90]),
+          ]
+
+print(donors)
+
+def thank_you():
+
+    selection = ""
+
+    print("== Send a Thank You ==\n")
+
+    print("Please enter the name of the person you would like to send a Thank You to."
+          "(type 'list' if you would like a list of the donor names\n")
+
+    response_name = input(">>")
+
+    if response_name == "list":
+        print(donors)
+
+    # Checking to see if the name is in the donors list
+    for item in donors:
+        if response_name in item:
+
+            selection = item
+            print(selection)
+
+    # If selection did not change from original value, then there is no match
+    if selection == "":
+
+        donors.append((response_name, []))
+
+    # Ask for donation amount; add donation amount to associated donation list
+    print("Please enter the donation amount:\n")
+
+    response_donation = input(">>")
+
+    for item in donors:
+        if response_name in item:
+           donors[donors.index(item)][1].append(response_donation)
+
+
+    print(donors)
+
+
+def report():
+    print("==. Generate a Report ==\n")
+
+
+
+def quit():
+    print("Are you sure you want to quit?\n"
+          "1) Yes, quit"
+          "2) No, don't quit")
+
+    response = input(">")
+
+    if response == "1":
+        sys.exit()
+    elif response == "2":
+        mainloop()
```

Comment:
careful here -- this will call the mainloop recursivly:

mainloop() calls quit(), quit() calls mainloop(), mainloop calls quit() .... etc, potentially a LOT of times.

so better for quit() to return -- then make sure that mainloop() will go back through the loop when it does.